packages/utils/scaffold.ts

```diff
@@ -45,16 +68,16 @@ export default function runTransform(transformConfig: TransformConfig, action: s
 
 	webpackConfig.forEach(
 		(scaffoldPiece: string): Promise<void> => {
-			const config: Config = transformConfig[scaffoldPiece];
+			let config: Config = transformConfig[scaffoldPiece];
 
-			const transformations = mapOptionsToTransform(config);
+			let transformations = mapOptionsToTransform(config);
 
 			if (config.topScope && transformations.indexOf("topScope") === -1) {
 				transformations.push("topScope");
 			}
 
-			if (config.merge) {
-				transformations.push("merge");
+			if (config.merge && transformations.indexOf("merge") === -1) {
+				[config, transformations] = mergeHandler(config, transformations);
```

Comment:
This is logic relevant to the generator providing the code, not the ast transform, this would make the abstraction bad. The generator is responsible to provide:

1. an array of indices with relevant data
2. The topScope property with the path

In the `handleMerge` function, you'll need to check if the part of the require is present in the most outerscope and append the second index if not present. 

Sorry for giving you an indication of what the easiest choice would be, it was a stepping stone towards a final solution

Reply:
To give you a hint: We're pushing the transformation, which means that it's probably the latest transform to be run, so what you'll need to do, is this logic, but directly as an ast transform. The data gets passed down by the each astTransform func

Reply:
Thanks for review! As far as I could understand now, I need to update `parseMerge` function directly in `ast-utills` 
to just use the 1st index of the array 
And push to the topScope here itself. Right?

Reply:
aye